ding/policy/plan_diffuser.py

```diff
@@ -0,0 +1,345 @@
+            
+            # step start update target model and frequence
+            step_start_update_target=2000,
+            update_target_freq=10,
+            # update weight of target net
+            target_weight=0.995,
+
+            value_step=200e3,
+
+            # (float) Weight uniform initialization range in the last output layer
+            init_w=3e-3,
+        ),
+    )
+
+    def default_model(self) -> Tuple[str, List[str]]:
+        return 'pd', ['ding.model.template.diffusion']
+    
+    def _init_learn(self) -> None:
+        r"""
+        Overview:
+            Learn mode init method. Called by ``self.__init__``.
+            Init q, value and policy's optimizers, algorithm config, main and target models.
+        """
+        # Init
+        self._priority = self._cfg.priority
+        self._priority_IS_weight = self._cfg.priority_IS_weight
+        self.action_dim = self._cfg.model.diffuser_model_cfg.action_dim
+        self.obs_dim = self._cfg.model.diffuser_model_cfg.obs_dim
+        self.n_timesteps = self._cfg.model.diffuser_model_cfg.n_timesteps
+        self.gradient_accumulate_every = self._cfg.learn.gradient_accumulate_every
+        self.plan_batch_szie = self._cfg.learn.plan_batch_size
+        self.gradient_steps = 1
+        self.update_target_freq = self._cfg.learn.update_target_freq
+        self.step_start_update_target = self._cfg.learn.step_start_update_target
+        self.target_weight = self._cfg.learn.target_weight
+        self.value_step = self._cfg.learn.value_step
+
+        # Optimizers
+        self._plan_optimizer = Adam(
+            self._model.diffuser.model.parameters(),
+            lr=self._cfg.learn.learning_rate,
+        )
+        self._value_optimizer = Adam(
+            self._model.value.model.parameters(),
+            lr=self._cfg.learn.learning_rate,
+        )
+
+        # Algorithm config
+        self._gamma = self._cfg.learn.discount_factor
+
+        # Main and target models
+        self._target_model = copy.deepcopy(self._model)
+        # self._target_model = model_wrap(
+        #     self._target_model,
+        #     wrapper_name='target',
+        #     update_type='momentum',
+        #     update_kwargs={'theta': self._cfg.learn.target_theta}
+        # )
+        self._learn_model = model_wrap(self._model, wrapper_name='base')
+        self._learn_model.reset()
+        # self._target_model.reset()
+
+        self._forward_learn_cnt = 0
+
+    def _forward_learn(self, data: dict) -> Dict[str, Any]:
+        loss_dict = {}
+
+        data = default_preprocess_learn(
+            data,
+            use_priority=self._priority,
+            use_priority_IS_weight=self._cfg.priority_IS_weight,
+            ignore_done=self._cfg.learn.ignore_done,
+            use_nstep=False
+        )
+        
+        conds = {}
+        val = data['condition_val']
+        id = data['condition_id']
+        for i in range(len(val)):
+            if id[i].item() in conds:
```

Comment:
you can first use `id = id.tolist()`, then directly use this list rather than the `item` operation.